### ISSUE TITLE
[CVE-2020-7731] CWE-476: NULL Pointer Dereference

### ISSUE THREAD

**Issue by johakoch** [russellhaering/gosaml2#99]
https://ossindex.sonatype.org/vulnerability/CVE-2020-7731?component-type=golang&component-name=github.com%2Frussellhaering%2Fgosaml2

This affects all versions of package github.com/russellhaering/gosaml2. There is a crash on nil-pointer dereference caused by sending malformed XML signatures.

**Comment by russellhaering** [comment]
You're right, but I'm not quite sure what is causing this. The Snyk report, which the CVE seems to reference, correctly notes that this was resolved in v0.7.0.

I've opened a [GitHub Security Advisory](https://github.com/russellhaering/gosaml2/security/advisories/GHSA-prjq-f4q3-fvfr) reflecting the status of this - perhaps that will propagate to the CVE?

I'll keep this open for tracking.

**Comment by johakoch** [comment]
Maybe, you could also [Report advisory or correction](https://ossindex.sonatype.org/doc/report-vulnerability) for ossindex.sonatype.org.
